### PR TITLE
[JS] Topic: Tutorial on Hoisting in Javascript

### LINKED ISSUE #161: JS: Hoisting in Javascript
# Hoisting in JavaScript
Discuss the concept of hoisting in JavaScript.

## Directory
```
Topics/6.Hoisting
```
## Description

- Talk about What is Hoisting
- How javascript uses hoisting for declarations
- Effect Of Hoisting on different types of Declarations

### PR DESCRIPTION
**Pull request by arp99**
<hr>

## Description 📜

Wrote a tutorial on Hoisting in Javascript
Fixes #161 
References:
- https://developer.mozilla.org/en-US/docs/Glossary/Hoisting
- https://www.programiz.com/javascript/hoisting
- https://kpealecodes.hashnode.dev/scoping-and-hoisting-in-javascript
- https://youtu.be/7XbfeMQyW4Q
- https://youtu.be/Fnlnw8uY6jo

<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev (Flutter, Java, Kotlin)
- [ ] Backend Dev (Java, .NET, Python)
- [ ] C/CPP
- [ ] Competitive Programming
- [ ] Cyber Security
- [ ] DSA
- [ ] Database
- [ ] Datascience with Python
- [ ] Datascience with R
- [ ] Frontend Dev (HTML, CSS, JS)
- [ ] Frontend Dev (React, Angular, Vue)
- [ ] Golang
- [ ] Interview Prep
- [ ] Java
- [x] JavaScript
- [ ] MERN
- [ ] Machine Learning
- [ ] Open Source
- [ ] Python
- [ ] Rust
- [ ] Statistics
- [ ] UI/UX

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

### CODE REVIEW

**Comment by arp99** [comment]
@paramsiddharth Please review my PR

File: Web_Development/Javascript/Topics/6.Hoisting/README.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+Obviously now you must have many doubts after reading the intro. So let's simplify it with an example.  
+Did you ever notice that you can call a function even before it is declared ? 
+```js
+    console.log(sayHi())
+    
+    function sayHi(){
+        console.log("Hi Girlscript!!")
+    }
+    // The result of the code is : Hi Girlscript!!
+```
```
Comment (paramsiddharth) [inline Web_Development/Javascript/Topics/6.Hoisting/README.md:17]: You don't need to indent the code-blocks. Please get rid of the indentation.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]
The rest looks alright to me. @Creatoon Please review this further!

File: Web_Development/Javascript/Topics/6.Hoisting/README.md (hunk 1 above)
Comment (arp99) [reply Web_Development/Javascript/Topics/6.Hoisting/README.md:17]: Ok I am removing the left indentations

Comment (arp99) [reply Web_Development/Javascript/Topics/6.Hoisting/README.md]: @paramsiddharth Added a commit to remove the indentation as per suggestions.

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi Please review/

**Review by paramsiddharth — APPROVED** [review]

**Review by dakshsethi — APPROVED** [review]

**Comment by dakshsethi** [comment]
Looks good to me!!

**Comment by paramsiddharth** [comment]
@Creatoon Please review.

**Review by Creatoon — APPROVED** [review]

**Pull request merged**